File: src/main/java/net/exoad/picflux/ui/AppUI.java
package net.exoad.picflux.ui;

import com.formdev.flatlaf.intellijthemes.FlatArcDarkIJTheme;
import javax.swing.*;
import lombok.extern.log4j.Log4j2;
@Log4j2
public class AppUI
{
	private AppUI()
	{
	}
	
	public static final class PublicConstants
	{
		private PublicConstants()
		{
		}
		
		public static final String K_WINDOW_TITLE="PicFlux";
		public static final int K_WINDOW_WIDTH=650;
		public static final int K_WINDOW_HEIGHT=410;
	}
	
	static
	{
		System.setProperty("sun.java2d.opengl","True");
		try
		{
			UIManager.setLookAndFeel(new FlatArcDarkIJTheme());
		}
		catch(UnsupportedLookAndFeelException e)
		{
			log.fatal("Unable to set look and feel: "+FlatArcDarkIJTheme.class);
			throw new RuntimeException(e);
		}
	}
	
	public static void composeMain()
	{
		new UIWindow()
			.withTitle(PublicConstants.K_WINDOW_TITLE)
			.withSize(
				PublicConstants.K_WINDOW_WIDTH,
				PublicConstants.K_WINDOW_HEIGHT
			).compose();
	}
}

File: src/main/java/net/exoad/picflux/ui/AppUI.kt
package net.exoad.picflux.ui

import com.formdev.flatlaf.intellijthemes.FlatArcDarkIJTheme
import javax.swing.JFrame
import javax.swing.UIManager
import javax.swing.UnsupportedLookAndFeelException
import jdk.vm.ci.hotspot.JFR
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.lang.RuntimeException

object AppUI
{
	private val log:Logger=LogManager.getLogger(AppUI::class.java)
	
	init
	{
		System.setProperty("sun.java2d.opengl" , "True")
		try
		{
			UIManager.setLookAndFeel(FlatArcDarkIJTheme())
		} catch (e:UnsupportedLookAndFeelException)
		{
			log.fatal(
				"Unable to set look and feel: {}" , FlatArcDarkIJTheme::class.java
			)
			throw RuntimeException(e)
		}
	}
	
	@JvmStatic
	@Synchronized
	fun composeMain()
	{
		JFrame().apply {
			title=PublicConstants.K_WINDOW_TITLE
			preferredSize=InternalUIHelper.dim(PublicConstants.K_WINDOW_WIDTH , PublicConstants.K_WINDOW_HEIGHT)
			size=preferredSize
			minimumSize=preferredSize
			
		}.run {
			setLocationRelativeTo(null)
			pack()
			requestFocus()
		}
	}
	
	object PublicConstants
	{
		const val K_WINDOW_TITLE:String="PicFlux"
		const val K_WINDOW_WIDTH:Int=650
		const val K_WINDOW_HEIGHT:Int=410
	}
}
